File: realtime/src/types/app_context.rs
use crate::config::Config;
use crate::services::AuthService;
use crate::types;

#[derive(Clone, Debug)]
pub struct Services {
    pub auth_service: AuthService,
}

#[derive(Clone)]
pub struct AppContext {
    pub amqp_channel: lapin::Channel,
    pub config: Config,
    pub meta: types::Meta,
    pub services: Services,
}

File: realtime/types/app_context.go
package types

import (
	amqp "github.com/rabbitmq/amqp091-go"

	"github.com/autobar-dev/services/realtime/repositories"
	"github.com/autobar-dev/shared-libraries/go/auth-repository"
)

type Repositories struct {
	Auth  *authrepository.AuthRepository
	State *repositories.StateRepository
	Mq    *repositories.MqRepository
}

type AppContext struct {
	MetaFactors  *MetaFactors
	Config       *Config
	Repositories *Repositories
	AmqpChannel  *amqp.Channel
}
